File: apps/admin-v2/src/infrastructure/api/repositories/api-onboarding.repository.ts
import type { OnboardingRepository, RegisterTenantData } from '@/domain/repositories/onboarding.repository';
import type { Tenant, BusinessType } from '@/domain/entities/tenant';
import api from '../client';
import { mapTenant } from '../mappers/tenant.mapper';

export class ApiOnboardingRepository implements OnboardingRepository {
  async register(data: RegisterTenantData): Promise<{ token: string; tenant: Tenant }> {
    const { data: res } = await api.post('/onboarding/register', {
      business_name: data.businessName,
      owner_name: data.ownerName,
      email: data.email,
      password: data.password,
    });
    return { token: res.token, tenant: mapTenant(res.tenant) };
  }

  async verify(code: string): Promise<void> {
    await api.post('/onboarding/verify', { code });
  }

  async checkSlug(slug: string): Promise<{ available: boolean }> {
    const { data } = await api.get('/onboarding/check-slug', { params: { slug } });
    return { available: data.available };
  }

  async setBusinessType(type: BusinessType, createServices: boolean): Promise<void> {
    await api.post('/onboarding/business-type', {
      business_type: type,
      create_services: createServices,
    });
  }
}

File: apps/admin_mobile/lib/infrastructure/api/repositories/api_onboarding_repository.dart
import '../../../domain/entities/tenant.dart';
import '../../../domain/repositories/onboarding_repository.dart';
import '../dio_client.dart';
import '../mappers/tenant_mapper.dart';

class ApiOnboardingRepository implements OnboardingRepository {
  final DioClient _client;

  ApiOnboardingRepository(this._client);

  @override
  Future<({String token, Tenant tenant})> register({
    required String businessName,
    required String ownerName,
    required String email,
    required String password,
  }) async {
    final response = await _client.dio.post('/onboarding/register', data: {
      'business_name': businessName,
      'owner_name': ownerName,
      'email': email,
      'password': password,
    });
    final data = response.data['data'] ?? response.data;
    return (
      token: data['token'] as String,
      tenant: mapTenant(data['tenant'] as Map<String, dynamic>),
    );
  }

  @override
  Future<void> verify(String code) async {
    await _client.dio.post('/onboarding/verify', data: {'code': code});
  }

  @override
  Future<bool> checkSlug(String slug) async {
    final response =
        await _client.dio.get('/onboarding/check-slug', queryParameters: {
      'slug': slug,
    });
    final data = response.data['data'] ?? response.data;
    return data['available'] as bool;
  }

  @override
  Future<void> setBusinessType(String type, bool createServices) async {
    await _client.dio.post('/onboarding/business-type', data: {
      'type': type,
      'create_services': createServices,
    });
  }
}
